src/kolibri_gnome/kolibri_context.py

```diff
@@ -289,13 +297,50 @@ def __kolibri_daemon_on_notify_is_started(
         if not self.__kolibri_daemon.props.is_started:
             return
 
-        # TODO: Check (via Kolibri's http API) if we need to do initial setup.
-        #       If we do, emit "requires-user-interaction".
-        #       Otherwise, set self.props.is_facility_ready = True
+        self.__kolibri_daemon.kolibri_api_get_async(
+            "/api/public/v1/facility/",
+            result_cb=self.__on_kolibri_api_facility_response,
+        )
+
+    def __on_kolibri_api_facility_response(self, data: list):
+        if len(data):
+            self.props.is_facility_ready = True
+            return
+
+        # There is no facility, so automatically provision the device:
+        self.__automatic_device_provision()
+
+    def __automatic_device_provision(self):
+        logger.info("Provisioning device…")
+        language_id = self.__get_closest_system_kolibri_language()
+        request_body_data = {
+            "language_id": language_id,
+            "facility": {"name": "Kolibri at home"},
+            "preset": "nonformal",
+            "superuser": {
+                "username": "admin",
+                "password": "admin",
+            },
```

Comment:
We should definitely not _release_ this (and close the ticket about it) until we can create a facility without a superuser, but depending on how much trouble that looks like, I'm happy merging branch this as-is so we can test it.

```suggestion
            # FIXME: Kolibri's API requires this. We should create a facility without a superuser.
            "superuser": {
                "username": "admin",
                "password": "admin",
            },
```

Reply:
Actually, I beat you to it. I pushed a couple commits which should solve this, and opened learningequality/kolibri#8798 :)

Reply:
Fantastic! Yes, let's not release this with the dumb admin/admin please.